### PR TITLE
Improve traffic and error capturing

### PR DESCRIPTION
**Pull request by schehata**
This PR provides 2 methods to facilitate logging HTTP requests.

The docs and examples have been updated to:
- Ask the users to create `middleware.ts` so that HTTP requests could be captured
- Encourage the users to create `error.tsx` to capture routing errors.

This will introduce new sources:
- middleware
- error.tsx

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,7 +47,28 @@ no further configuration is required.
...
+import { Logger } from 'next-axiom'
+import { NextResponse } from 'next/server'
+import type { NextFetchEvent, NextRequest } from 'next/server'
+
+export async function middleware(request: NextRequest, event: NextFetchEvent) {
+    const logger = new Logger({ source: 'middleware' }); // traffic, request
+    logger.middleware(request)
+
+    event.waitUntil(logger.flush())
+    return NextResponse.next()
```
Comment (bahlo) [inline README.md:65]: missing }

File: README.md [hunk 2]
```diff
@@ -47,7 +47,28 @@ no further configuration is required.
...
+
+Create a `middleware.ts` in the root dir of your app:
+
+```typescript
+import { Logger } from 'next-axiom'
+import { NextResponse } from 'next/server'
+import type { NextFetchEvent, NextRequest } from 'next/server'
+
+export async function middleware(request: NextRequest, event: NextFetchEvent) {
+    const logger = new Logger({ source: 'middleware' }); // traffic, request
```
Comment (bahlo) [inline README.md:61]: what does the comment mean?

File: README.md [hunk 3]
```diff
@@ -47,7 +47,28 @@ no further configuration is required.
...
+    logger.middleware(request)
+
+    event.waitUntil(logger.flush())
+    return NextResponse.next()
+
+
+// See "Matching Paths" below to learn more
```
Comment (bahlo) [inline README.md:68]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -157,6 +178,57 @@ const logger = new Logger({
     console.log(event.message);
   },
 });
+
+
+### Capturing Errors
+
+To capture routing errors we can use the [Error Handling](https://nextjs.org/docs/app/building-your-application/routing/error-handling) mechanism of Next. 
+
+Create a new file named `error.tsx` under your `/app` directory. Inside your component function use the logger to ingest the error to Axiom. e.g:
```
Comment (bahlo) [inline README.md:187]:
```suggestion
Create a new file named `error.tsx` under your `/app` directory. Inside your component function use the logger to ingest the error to Axiom. 

Example:
```

File: README.md [hunk 5]
```diff
@@ -157,6 +178,57 @@ const logger = new Logger({
...
+    <div className="p-8">
+      Ops! An Error has occurred:{" "}
+      <p className="text-red-400 px-8 py-2 text-lg">`{error.message}`</p>
+      <div className="w-1/3 mt-8">
+        <NavTable />
+      </div>
+    </div>
+  );
+}
 ```
```
Comment (bahlo) [inline README.md:232]: I think we need to write "create or edit" so it's clear that you can still use middleware and errors.tsx the way you did before

**Review by bahlo — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (schehata) [reply README.md]: it was the options we could use instead of `source: "middleware"`. I forgot to remove them. now removed.

File: README.md (hunk 1 above)
Comment (schehata) [reply README.md:65]: added

File: README.md (hunk 5 above)
Comment (schehata) [reply README.md:234]: done

**Pull request merged**